functions/venue.js

```diff
@@ -550,6 +551,10 @@ exports.updateVenue = functions.https.onCall(async (data, context) => {
     updated.radioStations = [data.radioStations];
   }
 
+  if (typeof data.showNametags === "boolean") {
+    updated.showNametags = data.showNametags;
+  }
+
   updated.requiresDateOfBirth = data.requiresDateOfBirth || false;
```

Comment:
[OOS] @denisdimitrov @sparkletown/engineering Is this a bug? I would have expected that it would need to follow the same `typeof` pattern that the other `boolean`s do here?

If not, do you know why it's different from the rest?

Reply:
[OOS] @0xdevalias This particular line will work correctly because there is a fallback value and if `requiresDateOfBirth` is `undefined` this will fallback to `false` and the firebase will accept it. If we have a fallback value there is no need for the `typeof` syntax or `if` statements at all.

The newly added code above can be changed to 
```tsx
updated.showNametags = data.showNametags || false
```
and it would still work correctly in every single case.

My suggestion is to replace all of the `typeof` checks with `updated.value = data.value || FALLBACK_VALUE` that way we can get rid of all of the `if` statements. This can be added as a future PR with clean up or something like that.

Keep in mind that some of the other fields are used only in certain venue templates and they should be in a switch case and added only for that venue template. Iframes, columns, rows, showGrid, etc...

Reply:
Simply removing the `typeof` and keeping the `if` statement wouldn't work either because the falsy values won't pass the statement, Ex: `false` for boolean, `""` for string, `0` for number and etc.

```tsx
if (updated.showSomething) {
updated.showSomething= data.showSomething
}
```

If `updated.showSomething` is `false` the statement will skip and the value in the firebase will remain `true`

@0xdevalias You probably already saw this in one of my latest pull requests, I'm commenting again just in case someone missed it or was wondering why use `typeof`

Reply:
Cool. Thanks for the insight here :)